src/EnergyPlus/ConvectionCoefficients.cc

```diff
@@ -5478,8 +5475,8 @@ void DynamicIntConvSurfaceClassification(EnergyPlusData &state, int const SurfNu
 
         // Calculate Grashof, Reynolds, and Richardson numbers for the zone
         // Grashof for zone air based on largest delta T between surfaces and zone height
-        Tmin = minval(TH(2, 1, {Zone(ZoneNum).HTSurfaceFirst, Zone(ZoneNum).HTSurfaceLast}));
-        Tmax = maxval(TH(2, 1, {Zone(ZoneNum).HTSurfaceFirst, Zone(ZoneNum).HTSurfaceLast}));
+        Tmin = minval(state.dataHeatBalSurf->SurfInTempHistCurr({Zone(ZoneNum).HTSurfaceFirst, Zone(ZoneNum).HTSurfaceLast}));
+        Tmax = maxval(state.dataHeatBalSurf->SurfInTempHistCurr({Zone(ZoneNum).HTSurfaceFirst, Zone(ZoneNum).HTSurfaceLast}));
```

Comment:
If you are looking for both min and max in a given range, it is faster to look for both of them together than to look for them separately. 

```
for (SurfNum = Zone(ZoneNum).HTSurfFirst; SurfNum <= Zone(ZoneNum).HTSurfLast; ++SurfNum)  {
    Real64 SurfTemp = SurfInTempCurr(SurfNum); 
    if (SurfTemp < Tmin) Tmin = SurfTemp;
    else if (SurfTemp > Tmax) Tmin = SurfTemp;
}
```

Not only are you traversing the elements once instead of twice, but you are also only checking for `Tmax` those elements that are greater than `Tmin`.